### ISSUE TITLE
Aliases on the base table using SQLSelect are lost

### ISSUE THREAD

**Issue by ScopeyNZ** [silverstripe/silverstripe-framework#8917]
## Affected Version

4.x

## Description

Here's some code to illustrate the problem

```
$select = SQLSelect::create('*', ['MyTableAlias' => 'MyTable']);
$sql = $select->sql();
echo $sql; // SELECT * FROM MyTable
```

## Steps to Reproduce

You can write the above code verbatim. I've got a PR (#8918) that illustrates the problem:

```
1) SilverStripe\ORM\Tests\SQLSelectTest::testBaseTableAliases
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'SELECT * FROM "MyTable" AS "MyTableAlias"'
+'SELECT * FROM "MyTable"'
```

## Notes

This is pretty easy to track down in the code:

https://github.com/silverstripe/silverstripe-framework/blob/db610aaf3bef897a1f3b5c60a307ee72a5d2a499/src/ORM/Queries/SQLConditionalExpression.php#L313-L340

The base table is shifted off the front of all tables so the alias is lost (it's stored as the key of that array value).

**Comment by ScopeyNZ** [comment]
In the end I just fixed the issue with the PR I raised that had the failing test as an example. #TDD

**Comment by ScopeyNZ** [comment]
Fixed by #8918
